### PR TITLE
Keep symbolic expressions as is and minor bugfixes optimization system

### PR DESCRIPTION
**Pull request by Vaibhavdixit02**

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
add tests which catch this?

**Comment by Vaibhavdixit02** [comment]
Failure is unrelated, something's up with RecursiveArrayTools bounds?

File: test/optimizationsystem.jl [hunk 1]
```diff
@@ -283,3 +283,16 @@ end
 
     @test sol1.u ≈ sol2.u
 end
+
+@testset "#2323 keep symbolic exressions and xor condition on constraint bounds" begin
```
Comment (ChrisRackauckas) [inline test/optimizationsystem.jl:287]:
```suggestion
@testset "#2323 keep symbolic expressions and xor condition on constraint bounds" begin
```

**Comment by Vaibhavdixit02** [comment]
There is a circular dependency between https://github.com/SciML/Optimization.jl/pull/636 and this PR so the test won't pass. I'll merge this and finish that one up and then create releases such that both CIs start passing with compat bounds.

**Comment by ChrisRackauckas** [comment]
That's fine, I am trying to get BifurcationKit to tag so we can at least see nothing else broke. If

**Comment by ChrisRackauckas** [comment]
Wait, did you run locally? Only seeing 41 seconds of failure isn't a good sign 😅

**Comment by Vaibhavdixit02** [comment]
Yeah have run it locally. The 41 secs is the RecursiveArrayTools thing I mentioned above, the version doesn't get resolved

**Pull request merged**